### ISSUE TITLE
No longer works as the environment value is no longer available in github actions

### ISSUE THREAD

**Issue by doozMen** [fwal/setup-swift#153]
Please update to not require to set an environment value to the swift versions

**Comment by fwal** [comment]
Hi @doozMen!

I'm not sure what you're referring to, could you please elaborate and, if possible, provide an example of your workflow configuration?

**Comment by doozMen** [comment]
You code expects and env value to be set to the swift version. This environment values you would set with `export swift-version=5.3` is no longer supported in github actions. The script fails on it. Just try it and you will see.

**Comment by doozMen** [comment]
The thing I did was on Ubuntu. If you need more info ping me. I had a release today so was a bit focused on that. I used another approach for now with personal access tokens. But I prefer ssh

**Comment by fwal** [comment]
Which version of Ubuntu and Swift is the workflow targeting?

**Comment by fwal** [comment]
Oh I realize now what you're referring to 👍🏻

**Comment by fwal** [comment]
Seems like my tests and GitHub changelog notifications missed that one 😅

**Comment by SomeRandomiOSDev** [comment]
It looks like all that needs to be done to resolve this issue is to update `@actions/core` to 1.2.6 ([source](https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/)) and it looks like that's already been done on the `main` branch. Any timeframe on when this might be released?

**Comment by Nef10** [comment]
I've just tried the last master version, it does not work yet:

```
Run fwal/setup-swift@3ee9bca324032daea2d3c1a67fdc1cbed46a9a35
  with:
    swift-version: 5.2.0
/usr/bin/xcrun --toolchain swift 5.2 --run swift --version
Apple Swift version 5.3 (swiftlang-1200.0.29.2 clang-1200.0.30.1)
Target: x86_64-apple-darwin19.6.0
/usr/bin/xar -x -C /Users/runner/work/_temp/e3058127-4d84-458c-8398-1f5bb858d79f -f /Users/runner/work/_temp/93e7f7af-77e0-450b-851b-b4064d6b9d52
/usr/bin/tar xz -C /Users/runner/work/_temp/5f7ead15-6b35-4ee2-9868-126299987c10 -f /Users/runner/work/_temp/e3058127-4d84-458c-8398-1f5bb858d79f/swift-5.2-RELEASE-osx-package.pkg/Payload
Error: Unable to process command '::add-path::/Users/runner/hostedtoolcache/swift-macOS/5.2/x64/usr/bin' successfully.

Error: The `add-path` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
Error: Unable to process command '::set-env name=TOOLCHAINS::swift 5.2' successfully.
Error: The `set-env` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
/Users/runner/hostedtoolcache/swift-macOS/5.2/x64/usr/bin/swift --version
Apple Swift version 5.2 (swift-5.2-RELEASE)
```

**Comment by fwal** [comment]
Hi everyone. Thank you for your patience, I haven't been able to work as much as I wanted on this and frankly I had no idea this was even used by more than one or two people so I'm humbled.

For the issue at hand I've pushed a preview for the next version with the tag `v1-next`, please try it out and let me know if it works for you.

Additionally I've started working on additional safeguards to prevent issues like this in the future.

**Comment by fwal** [comment]
Released 1.4.0 with the updated actions package and tags updated accordingly.
